src/sm/Materials/LatticeMaterials/latticematstatus.h

```diff
@@ -50,10 +50,16 @@ class NonlocalMaterialStatusExtension;
  * In this class services are defined that are used by other
  * lattice material statuses.
  */
-class LatticeMaterialStatus : public StructuralMaterialStatus, public RandomMaterialStatusExtensionInterface
+class LatticeMaterialStatus : public MaterialStatus, public RandomMaterialStatusExtensionInterface
 {
 protected:
 
+    FloatArray latticeStrain;
+    FloatArray tempLatticeStrain;
+
+    FloatArray latticeStress;
+    FloatArray tempLatticeStress;
```

Comment:
I would strongly recommend to make these fixed size `FloatArrayF<6>`. Then, only implementing 3D mode in all material models, and rely on the default iterative approach for 2D. It will be very very fast, even if it was to do a few iterations (or 1, in the case of a linear material). With fixed size arrays throughout the material models, one can avoid cache misses, enable much more optimizations.

Reply:
So, by this recommendation, I really think implementing 2D for the elastic/damage models, if it means having to use (dynamic, heap-allocated) FloatArray instead of FloatArrayF<6>, would be a big pessimization for all lattice models, including the elastic ones.

The iterative approach on that can utilize fixed size arrays is better in all aspects.

Reply:
I am happy with this. For elastic and scalar damage no iterative approach is needed. So, I just call the 3D case with zeros for the additional entries.